docs/delegate.md

```diff
@@ -0,0 +1,97 @@
+# Delegation
+
+For a more flexible configuration, you can have `server_name`
+resources (eg: `@user:example.com`) served by a different host and
+port (eg: `synapse.example.com:443`).
+
+Without configuring delegation, the matrix federation will
+expect to find your server via `example.com:8448`. The following methods
+allow you retain a `server_name` of `example.com` so that your user IDs, room
+aliases, etc continue to look like `*:example.com`, whilst having your
+federation traffic routed to a different server (e.g. `synapse.example.com`).
+
+## .well-known delegation
+
+To use this method, you need to be able to alter the
+`server_name` 's https server to serve the `/.well-known/matrix/server`
+URL. Having an active server (with a valid TLS certificate) serving your
+`server_name` domain is out of the scope of this documentation.
+
+The URL `https://<server_name>/.well-known/matrix/server` should
+return a JSON structure containing the key `m.server` like so:
+
+```json
+{
+    "m.server": "<synapse.server.name>[:<yourport>]"
+}
+```
+
+In our example, this would mean that URL `https://example.com/.well-known/matrix/server`
+should return:
+
+```json
+{
+    "m.server": "synapse.example.com:443"
+}
+```
+
+Note, specifying a port is optional. If no port is specified, then it defaults
+to 8448.
+
+Most installations will not need to configure .well-known. However, it can be
```

Comment:
is this still true? I think when we wrote it we were imagining that SRV would be the default, but that doesn't seem to be the case.

I'd lose this paragraph.

Reply:
Oh good catch. I'll only remove half of it though, because the bit about which TLS certificate to use is still useful.